templates/main.bicep

```diff
@@ -0,0 +1,358 @@
+targetScope = 'resourceGroup'
+
+// General parameters
+param location string = resourceGroup().location
+param tenantId string = subscription().tenantId
+param odhsiWebApiName string = 'ohdsi-webapi'
+param dockerRegistryServer string = 'https://index.docker.io/v1'
+param dockerImageName string = 'ohdsi/webapi'
+param dockerImageTag string = '2.12.1'
+param keyvaultName string ='keyvault-${uniqueString(resourceGroup().id)}'
+
+// Postgres parameters
+param serverEdition string = 'GeneralPurpose'
+param skuSizeGB int = 32
+param dbInstanceType string = 'Standard_D2s_v3'
+param availabilityZone string = '3'
+param version string = '14'
+param postgresDatabaseName string = 'atlas_db'
+param postgresSchemaName string = 'webapi'
+param flywayBaselineVersion string = '2.2.5.20180212152023'
+// Users and passwords
+param postgresAdminUsername string = 'postgres_admin'
+param postgresWebapiAdminUsername string = 'ohdsi_admin_user'
+param postgresWebapiAppUsername string = 'ohdsi_app_user'
+@secure()
+param postgresAdminPassword string
+@secure()
+param postgresWebapiAdminPassword string
+@secure()
+param postgresWebapiAppPassword string
+// Connection strings
+var jdbcConnectionStringWebapiAdmin = 'jdbc:postgresql://${postgresServer.properties.fullyQualifiedDomainName}:5432/${postgresDatabaseName}?user=${postgresWebapiAdminUsername}&password=${postgresWebapiAdminPassword}&sslmode=require'
```

Comment:
It looks like everyplace that uses this JDBC connection string also has a separate property for the password and yet the password is also included here. Why?

Reply:
where else do you see this? I mean we can construct the jdbc in the bash script itself in a similar way we do here but I think this way is cleaner.

Reply:
We decided for now to leave it as is since this is used as a keyvault reference in all other places.